### ISSUE TITLE
What is outOfBandRecord?

### ISSUE THREAD

**Issue by rafmudaf** [WebFreak001/code-debug#140]
@WebFreak001 can you please explain the outOfBandRecord output variable?

`GDB -> App: {"outOfBandRecord":[]}`

I'm trying to debug a c++ project but the debug adapter quits during execution. When I turn on printCalls and showDevDebugOutput, I see the message above.

**Comment by WebFreak001** [comment]
`outOfBandRecord` is a message from gdb to code debug which isn't a reply to a command (such as console output or the program pausing/stopping). As there can be multiple it can be an array, so you have an empty array there which means there were no out of band records between the last command and this command. It doesn't have anything to do with a command result, it is purely events from gdb.

A command in gdb mi is basically built like this: 
```
zero or more [out-of-band-record]
[result]
(gdb)
```
where all of the lines above always have new lines terminating them.

So basically out of band records are always a part of a response, though they get processed earlier. It could be argued that empty out of band records should not be omitted and that it is a bug, but it simplified the implementation.

**Comment by rafmudaf** [comment]
@WebFreak001 thanks for the explanation
